### DISCUSSION TITLE
How do I build this from source?

### DISCUSSION THREAD

**Discussion by atyshka** [bcakmakoglu/vue-flow#971] [Q&A] [answered]
I am using this package as a dependency of a project, and I want to fork it and use the source version rather than the npm version. I've tried just installing it locally with `npm install ./vue-flow` but that doesn't work because it installs everything as a monorepo, and then my imports don't work.

**Comment by bcakmakoglu** [comment]
Is there a specific reason why you're attempting this?

**Reply by atyshka** [reply]
I want to add more features and possibly recontribute them via PR. Currently, I want to create clean-looking loops without having to use pathfindingedge.

I think the problem is that the package.json references dist/, but npm/pnpm doesn't compile the library before linking it to my project? My temporary solution is building vue-flow in a separate directory and then importing after building it. It's more clunky to separately compile but it works

**Reply by bcakmakoglu** [reply]
Mh... If you're using pnpm you can use *link*

```json
{
  "dependencies": {
    "lib": "link:../lib"
  }
}
```

Assuming you have some directory structure like this

```
/projects
	/vue-flow
	/my-project
```

You can link the core package by using this

```json
{
  "dependencies": {
    "@vue-flow/core": "link:../vue-flow/packages/core"
  }
}
```

**Reply by bcakmakoglu** [reply, marked as answer]
Also just go into the `vue-flow` directory and follow the development steps in the README file

Basically this

1. go into the directory
2. install dependencies (`pnpm i`)
3. build the packages (`pnpm build`)

You're done, if you've linked the package as detailed in the previous post, you should be good to go :)
If you're looking to be able to develop the vue-flow packages while also developing your project you might want to run `pnpm dev` instead, which will basically run `pnpm build` but with a watcher that rebuilds the *js* files (types are not rebuilt with a watcher, you'll have to rebuild those yourself when you made changes)

**Reply by bcakmakoglu** [reply]
Also: any PRs are appreciated so feel free to open them up if you feel like having added a feature that might be useful to a wider audience 😄
